src/Nethermind/Nethermind.Network/StaticNodes/StaticNodesManager.cs

```diff
@@ -49,33 +49,37 @@ public async Task InitAsync()
         }
 
         string data = await File.ReadAllTextAsync(staticNodesPath);
-        IEnumerable<string> nodes = INodeSource.ParseNodes(data);
-        var nodeCount = nodes.Count();
+        ISet<string> nodeSet = INodeSource.ParseNodes(data);
 
         if (_logger.IsInfo)
-            _logger.Info($"Loaded {nodeCount} static nodes from {Path.GetFullPath(staticNodesPath)}");
+            _logger.Info($"Loaded {nodeSet.Count} static nodes from {Path.GetFullPath(staticNodesPath)}");
 
-        if (nodeCount != 0)
+        if (nodeSet.Count != 0)
         {
             if (_logger.IsDebug) _logger.Debug($"Static nodes: {Environment.NewLine}{data}");
         }
 
-        List<NetworkNode> networkNodes = [];
+        List<NetworkNode> nodes = [];
```

Comment:
Why do we need this List, then we do ToDictionary, then we create ConcurrentDictionary

Can we just have ConcurrentDictionary in the first place?

Reply:
I think it is doing exactly this in the 2nd place (maybe we can unify that code?). And it isn't that big of a change.